python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1616,11 +1615,9 @@ def agg(self, func):
                 return result.iloc[:, 0]
 
         # drop the first level if we have a multiindex
-        if (
-            isinstance(result.columns, pd.MultiIndex)
-            and result.columns.nlevels > 1
-        ):
-            result.columns = result.columns.droplevel(0)
+        cols = result._data.to_pandas_index()
+        if result._data.multiindex and cols.nlevels > 1:
```

Comment:
`ColumnAcessor` has `nlevels` attr.
```suggestion
        if result._data.nlevels > 1:
```

Reply:
Good call.